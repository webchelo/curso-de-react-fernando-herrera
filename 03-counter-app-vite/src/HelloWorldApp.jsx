export const HelloWorldApp = () => {
    // document.createElement...
    return (
        <h1>Hola Mundo!</h1>
    );
}