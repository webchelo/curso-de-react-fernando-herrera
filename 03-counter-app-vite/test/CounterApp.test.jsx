import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Pruebas en <CounterApp />', () => {
    const value = 10
    test('debe hacer match con el snapshot', () => {
        const {container} = render(<CounterApp value={ value } />)
        expect(container).toMatchSnapshot();
    })
    test('debe mostrar el valor inicial de 100', () => {
        render(<CounterApp value={ 100 } />)
        expect(screen.getByText(100)).toBeTruthy();
    })
    test('debe incrementar con el boton +1', () => {
        render(<CounterApp value={ value } />)
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('11')).toBeTruthy();
    })
    test('debe decrementar con el boton -1', () => {
        render(<CounterApp value={ value } />)
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('9')).toBeTruthy();
    })
    test('debe funcionar el boton de reset', () => {
        render(<CounterApp value={ 855 } />)
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        //fireEvent.click(screen.getByText('reset'))
        //expect(screen.getByText(855)).toBeTruthy();
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))
        expect(screen.getByText(855)).toBeTruthy();
    })
})