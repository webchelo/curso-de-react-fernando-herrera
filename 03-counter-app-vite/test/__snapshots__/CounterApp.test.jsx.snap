// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
